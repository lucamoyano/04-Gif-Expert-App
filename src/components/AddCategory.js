import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = ({ target: { value } }) => {
        setInputValue( value );
    }

    const handleSubmit = (e) => {
        e.preventDefault(); //Prevenir que se refresque la pagina
        
        if( inputValue.trim().length > 2 ) {
            setCategories( c => [inputValue, ...c]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <input 
                type="text" 
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}