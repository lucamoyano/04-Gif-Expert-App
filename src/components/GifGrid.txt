import React from 'react'
import {GifGridItem} from './GifGridItem';
import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ( {category} ) => {

    //const [images, setImages] = useState([]);

    // useEffect(() => {
    //     getGifs( category )
    //         //.then( imgs => setImages( imgs ) ); 
    //         //Como es una funcion que envia su primer argumento como primer argumento a la funcion
    //         //Se puede utilizar de la siguiente manera:
    //         .then( setImages );
    // }, [category]); 
    // //Le pasamod un arreglo vacio como segundo argumento
    // //Para decirle que se ejecute una sola vez

    const { data, loading } = useFetchGifs();

    return (
        <>
        <h3>{category}</h3>
        <div className="card-grid">
            {/* {
                images.map( ({id, title}) => (
                        <li key={id}> {title} </li>
                    )
                )
            } */}

            {/* {
                images.map( img => (
                    // <GifGridItem img={ img } />
                    <GifGridItem 
                        key={img.id}
                        { ...img }  //Le paso las props mediante desetructuracion
                    />
                    )
                )
            } */}
        </div>
        </>
    )
}
