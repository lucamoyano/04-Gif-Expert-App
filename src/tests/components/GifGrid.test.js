import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs'); //Fingir llamadas al useFetchGifs


describe('Pruebas en componente GifGrid', () => {

    const category = 'Goku';

    test('debe mostrar el componente GifGrid correctamente', () => {
        //Falsear la data, para evitar el error del mock
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true 
        });

        const wrapper = shallow( <GifGrid category = {category} /> )
        
        expect( wrapper ).toMatchSnapshot();

    });

    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        //Falsear la data, cuando obtengo una respuesta
        const gifs = [{
            id: 'ABC',
            url: 'https://url.com/image.png',
            title: 'Test'
        }];
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false 
        });

        const wrapper = shallow( <GifGrid category = {category} /> )
        
        expect( wrapper ).toMatchSnapshot();
        //Evaluar si existen componentes
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

    })
    
    

})