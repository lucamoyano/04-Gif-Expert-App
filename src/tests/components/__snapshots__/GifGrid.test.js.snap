// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifGrid debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Test"
      url="https://url.com/image.png"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en componente GifGrid debe mostrar el componente GifGrid correctamente 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <p>
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
